Tools/TestWebKitAPI/Tests/WebKitCocoa/SiteIsolation.mm

```diff
@@ -1304,4 +1304,63 @@ HTTPServer server({
     Util::run(&done);
 }
 
+TEST(SiteIsolation, SetFocusedFrame)
+{
+    auto mainframeHTML = "<iframe id='iframe' src='https://domain2.com/subframe'></iframe>"_s;
+    HTTPServer server({
+        { "/mainframe"_s, { mainframeHTML } },
+        { "/subframe"_s, { ""_s } }
+    }, HTTPServer::Protocol::HttpsProxy);
+
+    auto navigationDelegate = adoptNS([TestNavigationDelegate new]);
+    [navigationDelegate allowAnyTLSCertificate];
+    auto configuration = server.httpsProxyConfiguration();
+    enableSiteIsolation(configuration);
+    auto webView = adoptNS([[TestWKWebView alloc] initWithFrame:NSMakeRect(0, 0, 800, 600) configuration:configuration]);
+    webView.get().navigationDelegate = navigationDelegate.get();
+
+    [webView loadRequest:[NSURLRequest requestWithURL:[NSURL URLWithString:@"https://domain1.com/mainframe"]]];
+    [navigationDelegate waitForDidFinishNavigation];
+
+    __block RetainPtr<WKFrameInfo> mainFrameInfo;
+    __block RetainPtr<WKFrameInfo> childFrameInfo;
+    __block bool done = false;
+    [webView _frames:^(_WKFrameTreeNode *mainFrame) {
+        mainFrameInfo = mainFrame.info;
+        childFrameInfo = mainFrame.childFrames.firstObject.info;
+        done = true;
+    }];
+
+    EXPECT_FALSE(mainFrameInfo.get()._isFocused);
+    EXPECT_FALSE(childFrameInfo.get()._isFocused);
+
+    done = false;
+    [webView evaluateJavaScript:@"document.getElementById('iframe').focus()" completionHandler:^(id value, NSError *error) {
+        done = true;
+    }];
+    Util::run(&done);
+
+    do {
+        [webView _frames:^(_WKFrameTreeNode *mainFrame) {
+            mainFrameInfo = mainFrame.info;
+            childFrameInfo = mainFrame.childFrames.firstObject.info;
+        }];
+        Util::runFor(0.1_s);
```

Comment:
This is a little strange.  I'd expect Util::run(&done); here and set done to true once the _frames block is called.  Or even better, a shared lambda like this:

auto getUpdatedFrameInfo = [&] {
    done = false;
    [webView _frames:^(_WKFrameTreeNode *mainFrame) {
        mainFrameInfo = mainFrame.info;
        childFrameInfo = mainFrame.childFrames.firstObject.info;
        done = true;
    }];
    Util::run(&done);
};

do {
    getUpdatedFrameInfo();
} while (!mainFrameInfo.get()._isFocused || childFrameInfo.get()._isFocused); // Also, this should be || instead of &&

evaluateJavaScript

another do loop